lib/volcanic/authenticator/v1/privilege.rb

```diff
@@ -22,13 +22,23 @@ def initialize(scope:, permission_id: nil, group_id: nil, allow:, **args)
         @allow = allow
       end
 
+      def <=>(other)
```

Comment:
I *think* this is returning the most specific privilege scope last with true after false.

this means that given the privileges:
```ruby
{ scope: 'vrn:*:*:jobs/*', allow: false },
{ scope: 'vrn:*:*:jobs/*?creator=me', allow: true },
{ scope: 'vrn:*:*:jobs/23', allow: true },
{ scope: 'vrn:*:*:jobs/*?created_before=tuesday', allow: false }
```

would return:
```ruby
{ scope: 'vrn:*:*:jobs/*', allow: false },
{ scope: 'vrn:*:*:jobs/23', allow: false },
{ scope: 'vrn:*:*:jobs/24', allow: true }
```

This *probably* isn't an issue, but it is worth noting that where the scopes are more specific the most important is the last one, but when they are equally specific the most important (deny) may not be last

Reply:
Would it be a good idea to list some more test cases - I'll add them to the project and we can be sure?

Reply:
https://volcanicuk.atlassian.net/wiki/spaces/ENG/pages/413892609/Auth+service has a lot of worked examples which include details of which privileges would be returned by the auth service and which should be ultimately applied and reasons why... Are more examples than those provided there needed?

Reply:
I'll pick off a few and add them in to be sure.